From: Jordan Justen <jljusten@gmail.com>
Date: Tue, 22 Aug 2023 15:03:11 -0700
Subject: configure.ac: Only check FEATURE_SH_STANDALONE if using busybox

For example, running bash instead of busybox produces:

$ PATH= /bin/bash -c busybox 2>&1
/bin/bash: line 1: busybox: No such file or directory

Thus, the grep for "not found" fails, and configure mistakenly thinks
that bash is busybox with the FEATURE_SH_STANDALONE set.

Signed-off-by: Jordan Justen <jljusten@gmail.com>
---
 configure.ac | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index efac125..7bf02e3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -381,7 +381,8 @@ AC_CHECK_FUNCS([strsignal posix_fallocate sysconf])
 AC_ARG_WITH(sandbox-shell, AS_HELP_STRING([--with-sandbox-shell=PATH],[path of a statically-linked shell to use as /bin/sh in sandboxes]),
   sandbox_shell=$withval)
 AC_SUBST(sandbox_shell)
-if test ${cross_compiling:-no} = no && test -n "${sandbox_shell}"; then
+if test ${cross_compiling:-no} = no && test -n "${sandbox_shell}" &&
+   test -z "${sandbox_shell%%*busybox*}"; then
   AC_MSG_CHECKING([whether sandbox-shell has the standalone feature])
   # busybox shell sometimes allows executing other busybox applets,
   # even if they are not in the path, breaking our sandbox
